src/mmqueue.ts

```diff
@@ -4,46 +4,19 @@ import {
 } from './proto/matchmaking_pb';
 import { Player } from './mmplayer';
 import { PlayerChannel } from './mmchannel';
-
-interface QueueEntry {
-    ply: Player;
-    entryTime: number;
-}
+import { Match, QueueEntry, MatchConfig } from './mmmatch';
 
 interface PlayerInfo {
     matchState: MatchingState;
     channel: PlayerChannel;
 }
 
-class MatchConfig {
-    numPlayers = 8;
-
-    confirmTimeout = 12 * 1000;
-}
-
-interface MatchRecord {
-    ip: string;
-    port: number;
-}
-
-class Match {
-    players: Player[];
-
-    confirmTimer: NodeJS.Timeout|null = null;
-
-    parameters: MatchRecord|null = null;
-
-    constructor(ql: QueueEntry[]) {
-        this.players = ql.map((el: QueueEntry) => el.ply);
-    }
-}
-
 export class MatchMakingQueue {
-    players: Record<number, PlayerInfo> = {};
+    players: Record<string, PlayerInfo> = {};
```

Comment:
Comment (feel free to just hit resolve and not fix anything): It might be worth abstracting out types such as the fact that the string in this record means the Player's UID. As in, define `type PlayerUID = string;` and then use that wherever you would be using the Player UID. This would also allow easier adjustments in the future if the player uid typing would change again (even though it probably won't).

Reply:
This is definitely a good idea. I've updated the player type to reflect this and will be sure to apply it when I write the session/token system.